ocaml/xapi/xapi_host.ml

```diff
@@ -566,9 +566,18 @@ let evacuate ~__context ~host =
     match plan with
     | Migrate host ->
         ( try
+            let options =
+              ("live", "true")
+              ::
+              ( if network <> Ref.null then
+                  [("network", Ref.string_of network)]
```

Comment:
We should assert here that the network is suitable to avoid migration failures half-way through an evacuation. So we should do similar checks as in `pool_migrate` to assert that the network has a PIF with IP configuration on each host in the pool.

Reply:
The calls `pool_migrate`, I don't see the point of replicating the assertion. Am i missing something?

Reply:
Should iterate `assert_can_attach_network_on_host` on all pool's hosts?

Reply:
Okay, I tried something, what do you think?

Reply:
I don't like too much mixing building a list and checking assertion at the same time.